draco/analysis/dayenu.py

```diff
@@ -534,6 +534,93 @@ def process(self, stream):
         return stream
 
 
+class ApplyDelayFilterHybridVis(task.SingleTask):
+    """Apply a previously saved filter to the hybrid beamformed visibilities.
+
+    This task takes a DAYENU filter saved in hybrid beamformed data and applies to
+    another hybrid beamformed visibilities. This task is used to apply the foreground
+    filter to the 21-cm simulation.
+
+    Attributes
+    ----------
+    atten_threshold : float
+        Mask any frequency where the diagonal element of the filter
+        is less than this fraction of the median value over all
+        unmasked frequencies.  Default is 0.0 (i.e., do not mask
+        frequencies with low attenuation).
+    """
+
+    atten_threshold = config.Property(proptype=float, default=0.0)
+
+    def setup(self, ss):
+        """Set the DAYENU filter to be applied to hybrid beamformed data.
+
+        Parameters
+        ----------
+        ss: containers.HybridVisStream
+          The filter of HybridVisStream to be applied.
+        """
+        self.sstream = ss
+
+    def process(self, hv):
+        """Apply the DAYENU filter to a HybridVisStream.
+
+        Parameters
+        ----------
+        hv: containers.HybridVisStream
+         The data the filter will be applied to.
+
+        Returns
+        -------
+        hv_filt: containers.HybridVisStream
+          The filtered dataset.
+        """
+        # Distribute over products
+        hv.redistribute(["ra", "time"])
+        self.sstream.redistribute(["ra", "time"])
+
+        npol, nfreq, new, nel, ntime = hv.vis.local_shape
+
```

Comment:
It might be a good idea to check that the various axes (freq, ew, pol, time) match between the filter and the hybrid-beamformed visibilities using `np.array_equal` similar to what is done at the start of `DeconvolveHybridMBase`.  This would guard against a situation where different subsets of the data have been loaded for the two containers.

Reply:
done